crypto/elliptic-curve.ts

```diff
@@ -58,7 +58,9 @@ const a = 0n;
 const projectiveZero = { x: 1n, y: 1n, z: 0n };
 
 type GroupProjective = { x: bigint; y: bigint; z: bigint };
-type GroupAffine = { x: bigint; y: bigint; infinity: boolean };
+type PointAtInfinity = { x: bigint; y: bigint; infinity: true };
+type FinitePoint = { x: bigint; y: bigint; infinity: false };
+type GroupAffine = PointAtInfinity | FinitePoint;
```

Comment:
isn't this the same type as before now? since `boolean = true | false`?

I thought the version where you had `type PointAtInfinity = { infinity: true };` made sense, but not sure how cumbersome it would be to use. If a lot of work, I'm also fine with keeping the type

Reply:
> I thought the version where you had type PointAtInfinity = { infinity: true }; made sense, but not sure how cumbersome it would be to use. If a lot of work, I'm also fine with keeping the type

This `type PointAtInfinity = { infinity: true };` was my original approach and made sense to me also. When I added this type, changes were necessary in many methods that used the `GroupAffine` type in the `eliptic-curve.ts` and `elliptic-curve-endomorphism.ts` files because those methods were destructuring `x,y, infinity` in the parameters.

I can easily revert to my previous approach and make the necessary changes in the codebase. I only refactored because it required the least amount of code changes.

Reply:
Ok, I think going with the least amount of changes is probably best!